cdap-api/src/main/java/co/cask/cdap/api/schedule/ScheduleConfigurer.java

```diff
@@ -32,9 +32,9 @@
 
   ScheduleConfigurer delayRun(long delayMillis);
 
-  ScheduleConfigurer setTimeRange(int startHour, int endHour);
+  ScheduleConfigurer setTimeRange(int startHour, int startMinute, int endHour, int endMinute);
```

Comment:
I have refactored this class after rebasing with your changes. It now is a little more elaborate: it will have startTime and endTime and timeZone as strings, to persist in JSON, and transient members for the stuff you added for the constraint checking. So this is not a concern any more